### ISSUE TITLE
CSS not generated server-side when using `renderToStaticOptimized` and `className`

### ISSUE THREAD

**Issue by colepeters** [threepointone/glamor#103]
## Relevant version info
Node v4 LTS
Glamor @ 2.17.14

## Problem
I’m in the early stages of writing a universal React app using Glamor. I’ve just spotted an issue where, when my components are styled via `merge()` in their `className` attribute, and I use `renderToStaticOptimized`, no styles are produced in my document’s `<style>` tag. However, if I make all the components spread the styles into their attributes (with `{..merge()}`), I get the styles rendered just fine.

I’d prefer to keep my styles within each component’s `className` attribute. Any idea why this wouldn’t be working?

## Example
Given this app setup:
```js
function render(Document, Component, applicationContext) {
  …
  const { html, css, ids } = renderStaticOptimized(() => renderToString(<RootComponent />));

  return `<!doctype html>${ renderToStaticMarkup(<Document title='foo' css={ css } ids={ ids }>{ html }</Document>) }`;
}
```

…and this top-level component:
```jsx
import React from 'react';

export default function Document({ title, css, children }) {
  const content = [
    …
  ].join('');

  return (
    <html lang='en-US' dir='ltr'>
      <head>
        <meta httpEquiv='Content-Type' content='text/html; charset=utf-8' />
        <title>{ title }</title>
        <meta name='viewport' content='width=device-width, initial-scale=1.0' />
        <style dangerouslySetInnerHTML={ { __html: css } } />
      </head>
      <body dangerouslySetInnerHTML={ { __html: content } } />
    </html>
  );
}
```

…which includes this component:
```jsx
import React from 'react';
import { style, merge } from 'glamor';

import { MAX_CONTAINER_WIDTH } from './constants';
import { large } from '../../styles/constants/mediaQueries';
import { mha } from '../../styles/margin';
import { pl2, pr2, pl3Md, pr3Md, pl4Lg, pr4Lg } from '../../styles/padding';

export default function Container(props) {
  const containerMaxWidth = style({
    …
  });

  return (
    <div className={ merge(mha, pl2, pr2, pl3Md, pr3Md, pl4Lg, pr4Lg, containerMaxWidth) }>
      { props.children }
    </div>
  );
};

Container.propTypes = {
  children: React.PropTypes.node.isRequired
};
```

…my `<style>` tag is empty:

* * *

<img src='https://cloud.githubusercontent.com/assets/1713932/20013370/c5568e7a-a2aa-11e6-9771-1bce939d9688.png' width='450px' border='1px solid black' />

* * *

However, if I change my component to use spread its style into its attributes, like so:
```jsx
import React from 'react';
import { style, merge } from 'glamor';

import { MAX_CONTAINER_WIDTH } from './constants';
import { large } from '../../styles/constants/mediaQueries';
import { mha } from '../../styles/margin';
import { pl2, pr2, pl3Md, pr3Md, pl4Lg, pr4Lg } from '../../styles/padding';

export default function Container(props) {
  …
  });

  return (
    <div { ...merge(mha, pl2, pr2, pl3Md, pr3Md, pl4Lg, pr4Lg, containerMaxWidth) }>
      { props.children }
    </div>
  );
};

Container.propTypes = {
  children: React.PropTypes.node.isRequired
};
```

…then I get my styles:

* * *

<img src='https://cloud.githubusercontent.com/assets/1713932/20013425/f7bca9b2-a2aa-11e6-99cd-d10e597b064d.png' width='450px' />

* * *

**Comment by threepointone** [comment]
odd, having a look

**Comment by threepointone** [comment]
looks like I had a bug. pushed a fix, please update to 2.17.15. sorry!

**Comment by colepeters** [comment]
Thank you!
